gammapy/datasets/core.py

```diff
@@ -455,6 +455,9 @@ def write(
             )
             data["datasets"].append(d)
 
+        if path.exists() and not overwrite:
```

Comment:
Maybe add a test for this?

Reply:
@QRemy added these lines, but I am actually not sure why. We already pick up on the overwrite?

When I try to test this it has an error at line 453. And the final error is actually from astropy. Maybe I am doing something wrong

Reply:
It added this so the overwrite option apply also to the datasets.yaml file otherwise it is always overwritten